crates/pallets/starknet/src/blockifier_state_adapter.rs

```diff
@@ -32,20 +35,26 @@ impl<T> StateChanges for BlockifierStateAdapter<T>
 where
     T: Config,
 {
-    fn count_state_changes(&self) -> (usize, usize, usize, usize) {
+    fn count_state_changes(
+        &self,
+    ) -> (UpdatedContractsCount, UpdatedStorageVarsCount, DeclaredClassesCount, DeclaredCompiledClassesCount) {
         let keys = self.storage_update.keys();
         let n_contract_updated = BTreeSet::from_iter(keys.clone().map(|&(contract_address, _)| contract_address)).len();
         (n_contract_updated, keys.len(), self.class_hash_update, self.compiled_class_hash_update)
     }
 }
 
-impl<T> FeeConfig for BlockifierStateAdapter<T>
+impl<T> StateConfigProvider for BlockifierStateAdapter<T>
 where
     T: Config,
 {
     fn is_transaction_fee_disabled(&self) -> bool {
         T::DisableTransactionFee::get()
     }
+
+    fn is_nonce_validation_disabled(&self) -> bool {
+        T::DisableNonceValidation::get()
+    }
 }
```

Comment:
This don't really have to be implemented on the BlockifierStateAdapter. It's logic used by the execution, not the state.

I think those should be read from the config before calling execute